### PR TITLE
Add platform.sh support

### PR DESCRIPTION
**Pull request by senghe**
Hi. I've already prepared a platform.sh setup configuration for Sylius. In a nearest future I'll push also a documentation for this feature into Sylius repository.

I hope you like it :vulcan_salute:

### CODE REVIEW

File: .env.platform.dist [hunk 1]
```diff
@@ -0,0 +1 @@
+DATABASE_URL=mysql://root@127.0.0.1/sylius_%kernel.environment%?charset=utf8mb4&serverVersion=mariadb-11.0.0
```
Comment (diimpp) [inline .env.platform.dist]: Is it necessary to have independent .env template for psh?

Comment (diimpp) [inline .env.platform.dist]: `serverVersion` doesn't match selected PSH database.

File: .node-version [hunk 2]
```diff
@@ -0,0 +1 @@
+16.20.2
```
Comment (diimpp) [inline .node-version]: Unless there is reason to have minimal minor version, better to select major version only to automatically get latest version.

```suggestion
16
```

Comment (diimpp) [reply .node-version]: I think 1.12 still on 16, but 1.13 on 18/20? :thinking:

File: .platform.app.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+name: sylius
+
+type: php:8.1
```
Comment (diimpp) [inline .platform.app.yaml]: I think it should be either minimal (8.0 for 1.12) version or latest version.

```suggestion
type: php:8.3
```

Comment (diimpp) [inline .platform.app.yaml]: Build flavor should be specified at the top.

```suggestion
type: php:8.1
build:
    flavor: none
```

File: .platform.app.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+relationships:
+    database: "db:mysql"
+
+disk: 2048
+
+mounts:
+    "/var/cache": "shared:files/cache"
```
Comment (diimpp) [inline .platform.app.yaml]: Missing `config/jwt`, otherwise it will fail to generate it on readonly FS.

File: .platform.app.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    "/var/log": "shared:files/log"
```
Comment (diimpp) [inline .platform.app.yaml]: To my knowledge, `shared:files/log` is outdated syntax, that can be seen in years old examples, but it's not a default syntax in docs anymore. https://docs.platform.sh/add-services/network-storage.html#2-add-the-relationship

```suggestion
    "/var/log": 
       source: local
       source_path: log
```

File: .platform.app.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    "/var/sessions": "shared:files/sessions"
+    "/public/uploads": "shared:files/uploads"
```
Comment (diimpp) [inline .platform.app.yaml]: Missing `'/public/bundles'` unless it somehow transferred at build step.

File: .platform.app.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    
+    deploy: |
+        set -e
+        symfony-deploy
+        rm -rf var/cache/*
+        mkdir -p public/media/image
+        bin/console sylius:install -n
+        bin/console sylius:fixtures:load -n
+        bin/console assets:install --symlink --relative public
+        bin/console cache:clear
```
Comment (diimpp) [inline .platform.app.yaml:112]: Missing 
> bin/console lexik:jwt:generate-keypair --skip-if-exists

File: .platform.app.yaml [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+hooks:
+    build: |
+        set -e 
+        cp .env.platform.dist .env.local
+        composer --no-ansi --no-interaction install --no-progress --prefer-dist --optimize-autoloader
+        curl -fs https://get.symfony.com/cloud/configurator | bash
+        n auto
+        hash -r
+        yarn install
+        GULP_ENV=prod yarn build
```
Comment (diimpp) [inline .platform.app.yaml]: I think 1.12 is using webpack by default?
```suggestion
        yarn install --frozen-lockfile
        yarn build:prod
```

File: .platform.app.yaml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        cmd: "php bin/console sylius:promotion:generate-coupons"
+    php bin/console sylius:remove-expired-carts:
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:cancel-unpaid-orders"
+        
+hooks:
+    build: |
+        set -e 
+        cp .env.platform.dist .env.local
+        composer --no-ansi --no-interaction install --no-progress --prefer-dist --optimize-autoloader
```
Comment (diimpp) [inline .platform.app.yaml]: Why not use composer build flavor?

File: .platform.app.yaml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        yarn install
+        GULP_ENV=prod yarn build
+    
+    deploy: |
+        set -e
+        symfony-deploy
+        rm -rf var/cache/*
+        mkdir -p public/media/image
+        bin/console sylius:install -n
+        bin/console sylius:fixtures:load -n
```
Comment (diimpp) [inline .platform.app.yaml]:
```suggestion
        bin/console sylius:install -n
        bin/console sylius:fixtures:load -n
        # {Smart comment placeholder :') }
        # doctrine:migration:migrate --no-interaction
```

File: .platform.app.yaml (hunk 3 above)
Comment (diimpp) [reply .platform.app.yaml]: Overall placement feels weird, I personally would go for

- type
- build
- runtime
- variables
- dependencies
- hooks
- relationships
- disk
- mounts
- web
- workers
- cron

or last three as
- workers
- cron
- web

File: .platform.app.yaml [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+web:
+    locations:
+        "/":
+            root: "public"
+            passthru: "/index.php"
+            allow: true
+            expires: -1
+            scripts: true
+        '/assets/shop':
```
Comment (diimpp) [inline .platform.app.yaml:44]: For webpack,
```suggestion
        '/build':
```

File: .platform.app.yaml [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+        N_PREFIX: /app/.global
+
+build:
+    flavor: none
+
+dependencies:
+    nodejs:
+        n: "*"
+        yarn: "*"
+        gulp-cli: "*"
```
Comment (diimpp) [inline .platform.app.yaml]:
```suggestion
```

File: .platform.app.yaml [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    php:
+        composer/composer: '^2'
+
+crons:
+    cancel-unpaid-orders:
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:cancel-unpaid-orders"
+    generate-promotion-coupons:
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:promotion:generate-coupons"
```
Comment (diimpp) [inline .platform.app.yaml]: Unless I'm mistaken, this command can't be used like so and needs to be removed.

image

```suggestion
```

File: .platform.app.yaml [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+dependencies:
+    nodejs:
+        n: "*"
+        yarn: "*"
+        gulp-cli: "*"
+    php:
+        composer/composer: '^2'
+
+crons:
+    cancel-unpaid-orders:
```
Comment (diimpp) [inline .platform.app.yaml]: `:` is not a valid char at worker name (even wrapped on ' '), but can be used at cron, but it doesn't sound like a good idea take advantage of this bug.

```suggestion
    'sylius:cancel-unpaid-orders':
```
```suggestion
    sylius-cancel-unpaid-orders:
```

File: .platform.app.yaml [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:cancel-unpaid-orders"
+    generate-promotion-coupons:
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:promotion:generate-coupons"
+    php bin/console sylius:remove-expired-carts:
```
Comment (diimpp) [inline .platform.app.yaml]:
```suggestion
    sylius-remove-expired-carts:
```

File: .platform.app.yaml [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:cancel-unpaid-orders"
```
Comment (diimpp) [inline .platform.app.yaml]: Not sure, that cron commands get `APP_ENV=prod`, better be safe, than sorry :smiley: 

```suggestion
        cmd: "php bin/console --env=prod sylius:cancel-unpaid-orders"
```

File: .platform.app.yaml [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+        n: "*"
+        yarn: "*"
+        gulp-cli: "*"
+    php:
+        composer/composer: '^2'
+
+crons:
+    cancel-unpaid-orders:
+        spec: "0 2 * * *"
+        cmd: "php bin/console sylius:cancel-unpaid-orders"
```
Comment (diimpp) [inline .platform.app.yaml]:
```suggestion
        cmd: "php bin/console --env=prod sylius:cancel-unpaid-orders"
```

**Review by diimpp — DISMISSED** [review]

File: .platform.app.yaml (hunk 5 above)
Comment (diimpp) [reply .platform.app.yaml]: Personally I would remove `files` part from `source_path` as each definition is a separate mount and there is no point prefixing them.

File: .platform.app.yaml (hunk 3 above)
Comment (TheMilek) [reply .platform.app.yaml]: I will leave it for now as it is similar to config provided for the template by platform.sh 🤔 https://github.com/platformsh-templates/sylius/blob/master/.platform.app.yaml

File: .platform.app.yaml (hunk 6 above)
Comment (TheMilek) [reply .platform.app.yaml]: seems as it is not needed, works without it
edit: nevermind you were right with the bundles, the assets were not installing 😅

**Review by mpysiak — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Marcin! :1st_place_medal:

**Pull request merged**